src/OpenTelemetry.Exporter.Jaeger/Implimentation/InMemoryTransport.cs

```diff
@@ -0,0 +1,101 @@
+// <copyright file="InMemoryTransport.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Exporter.Jaeger.Implimentation
+{
+    using System;
+    using System.IO;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Thrift.Transports;
+
+    internal class InMemoryTransport : TClientTransport
+    {
+        private readonly MemoryStream byteStream;
+        private bool isDisposed;
+
+        public InMemoryTransport()
+        {
+            this.byteStream = new MemoryStream();
+        }
+
+        public override bool IsOpen => true;
+
+        public override async Task OpenAsync(CancellationToken cancellationToken)
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                await Task.FromCanceled(cancellationToken);
```

Comment:
No need to make method `async`.
```suggestion
                return Task.FromCanceled(cancellationToken);
```